File: appPlatform/src/main/java/com/app/appplatform/entity/StoreLinkConfig.java
package com.app.appplatform.entity;

import lombok.Data;

import java.util.Date;

/**
 * 应用商店链接配置实体类
 */
@Data
public class StoreLinkConfig {
    /**
     * 主键ID
     */
    private Long id;

    /**
     * 设备品牌（如：xiaomi、huawei、honor、oppo等）
     */
    private String deviceBrand;

    /**
     * 应用商店链接模板（如：market://details?id={packageName}）
     */
    private String linkTemplate;

    /**
     * 是否启用（1=启用，0=禁用）
     */
    private Integer enabled;

    /**
     * 排序权重
     */
    private Integer sortOrder;

    /**
     * 备注
     */
    private String remark;

    /**
     * 创建时间
     */
    private Date createTime;

    /**
     * 更新时间
     */
    private Date updateTime;
}

File: appplatform-manager/src/api/storeLinkConfig.ts
import request from "@/utils/request";

const prefix = "/api/store-link-config";

export interface StoreLinkConfig {
  id: number;
  deviceBrand: string;
  linkTemplate: string;
  enabled: number; // 1 for enabled, 0 for disabled
  sortOrder: number;
  remark: string;
  createTime?: string;
  updateTime?: string;
}

export interface StoreLinkConfigDto {
  deviceBrand: string;
  linkTemplate: string;
  enabled: number;
  sortOrder: number;
  remark?: string;
}

// 获取所有配置
export const getAllConfigs = () => {
  return request.get<StoreLinkConfig[]>(`${prefix}`);
};

// 根据ID查询配置
export const getConfigById = (id: number) => {
  return request.get<StoreLinkConfig>(`${prefix}/${id}`);
};

// 创建配置
export const createConfig = (data: StoreLinkConfigDto) => {
  return request.post<StoreLinkConfig>(`${prefix}`, data);
};

// 更新配置
export const updateConfig = (id: number, data: StoreLinkConfigDto) => {
  return request.put<StoreLinkConfig>(`${prefix}/${id}`, data);
};

// 删除配置
export const deleteConfig = (id: number) => {
  return request.delete<boolean>(`${prefix}/${id}`);
};

// 启用/禁用配置
export const updateConfigEnabled = (id: number, enabled: number) => {
  return request.put<boolean>(`${prefix}/${id}/enabled`, null, {
    params: { enabled },
  });
};

const appPrefix = "/api-app";

// 获取应用更新总开关状态
export const getAppUpdateStatus = () => {
  return request.get<{ appUpdateTotalEnabled: boolean }>(
    `${appPrefix}/update-total/status`,
  );
};

// 设置应用更新总开关状态
export const setAppUpdateStatus = (enabled: boolean) => {
  return request.post<string>(`${appPrefix}/update-total/set-status`, null, {
    params: { enabled },
  });
};
